public/src/overmind/actions.tsx

```diff
@@ -382,45 +367,60 @@ export const loadCourseSubmissions = async ({ state, actions }: Context, courseI
 
 /** Refreshes all submissions for a given course. Calling this action directly will not trigger the loading spinner.
  *  Use `loadCourseSubmissions` instead if you want to trigger the loading spinner, such as on page load. */
-export const refreshCourseSubmissions = async ({ state, actions, effects }: Context, courseID: bigint): Promise<void> => {
+export const refreshCourseSubmissions = async ({ state, effects }: Context, courseID: bigint): Promise<void> => {
     // None of these should fail independently.
-    const result = await effects.grpcMan.getSubmissionsByCourse(courseID, SubmissionRequest_SubmissionType.USER)
-    const groups = await effects.grpcMan.getSubmissionsByCourse(courseID, SubmissionRequest_SubmissionType.GROUP)
-    if (!success(result) || !success(groups)) {
-        const failed = !success(result) ? result : groups
-        actions.alertHandler(failed)
-        return
-    }
-    if (result.data) {
-        state.submissionsForCourse.setSubmissions("USER", result.data)
-        for (const submissions of Object.values(result.data.submissions)) {
-            for (const submission of submissions.submissions) {
-                state.review.reviews.set(submission.ID, submission.reviews)
-            }
+    const userResponse = await effects.api.client.getSubmissionsByCourse({
+        CourseID: courseID,
+        FetchMode: {
+            case: "Type",
+            value: SubmissionRequest_SubmissionType.USER
+        }
+    })
+    const groupResponse = await effects.api.client.getSubmissionsByCourse({
+        CourseID: courseID,
+        FetchMode: {
+            case: "Type",
+            value: SubmissionRequest_SubmissionType.GROUP
         }
+    })
+    if (userResponse.error || groupResponse.error) {
```

Comment:
What will happen if the first query worked and the second failed? Should this instead be two separate `if (xxxResponse.error)` checks, so that the second call will not be called if the first failed? I'm not sure what is the right semantics here.